rbc/omnisci_backend/numpy_ufuncs.py

```diff
@@ -9,7 +9,17 @@
 
 
 def determine_dtype(a, dtype):
-    return a.eltype if dtype is None else dtype
+    if isinstance(a, ArrayPointer):
+        return a.eltype if dtype is None else dtype
+    else:
+        return a if dtype is None else dtype
+
+
+def determine_input_type(argty):
+    if argty == typesystem.boolean8:
+        return bool
+    else:
+        return argty.eltype if isinstance(argty, ArrayPointer) else argty
```

Comment:
What are the allowed return values of `determine_dtype` and `determine_input_type`? Python types, Numba types, anything else? I presume to return typesystem types is not allowed.

Reply:
`determine_dtype` and `determine_input_type` must return numba types.

* `determine_input_type`: Only exists to explicitly convert `boolean8` values to `bool`. Without this, one would have to write numpy functions to work with `boolean8` args (see #146). 
* `determine_dtype`: Explicit convert the output to a dtype.

> I presume to return typesystem types is not allowed.

Yes, you're right.